src/stronghold/mod.rs

```diff
@@ -579,13 +626,16 @@ pub async fn generate_address(
     check_snapshot(&mut runtime, snapshot_path, None).await?;
     load_private_data_actor(&mut runtime, snapshot_path, None).await?;
 
-    let chain = Chain::from_u32_hardened(vec![
+    let chain = [
```

Comment:
Why not `use crypto::keys::bip44`? This way you can use `bip44::Bip44::from([4218, account_index, internal, address_index])` to represent BIP44 chain. It's easier and safer to use, you don't have to manually `.into_iter().map(|s| Segment::harden(s).into()).collect()`.

Reply:
Hm but Stronghold requires `type SLIP10Chain = Vec<u32>` and if I use `Bip44::.to_chain::<ed25519::SecretKey>()` I get this mismatch errror
```
mismatched types
expected struct `Vec<u32>`
    found array `[crypto::keys::slip10::Hardened; 5]
```

Reply:
Hm but Stronghold requires `type SLIP10Chain = Vec<u32>` and if I use `Bip44::.to_chain::<ed25519::SecretKey>()` I get this mismatch errror
```
mismatched types
expected struct `Vec<u32>`
    found array `[crypto::keys::slip10::Hardened; 5]
```

Reply:
Right. We should have used Bip44 in stronghold Slip10 API. With `Vec<u32>` it is more generic but less convenient.

Reply:
It's updated for the new stronghold version, but will not happen in this version and wallet.rs anymore.